digitiser-aggregator/src/main.rs

```diff
@@ -83,33 +90,44 @@ async fn main() {
 
     let ttl = Duration::from_millis(args.frame_ttl_ms);
 
-    let mut cache = FrameCache::<EventData>::new(ttl, args.digitiser_ids);
+    let mut cache = FrameCache::<EventData>::new(ttl, args.digitiser_ids.clone());
 
     let mut cache_poll_interval = tokio::time::interval(Duration::from_millis(args.cache_poll_ms));
     loop {
         tokio::select! {
             event = consumer.recv() => {
                 match event {
                     Ok(msg) => {
-                        on_message(&mut cache, &producer, &args.output_topic, &msg).await;
+                        on_message(&root_span, &args, &mut cache, &producer, &msg).await;
                         consumer.commit_message(&msg, CommitMode::Async).unwrap();
                     }
                     Err(e) => warn!("Kafka error: {}", e),
                 };
             }
             _ = cache_poll_interval.tick() => {
-                cache_poll(&mut cache, &producer, &args.output_topic).await;
+                cache_poll(&args, &mut cache, &producer).await;
             }
         }
     }
 }
 
 async fn on_message(
+    root_span: &Span,
+    args: &Cli,
     cache: &mut FrameCache<EventData>,
     producer: &FutureProducer,
-    output_topic: &str,
     msg: &BorrowedMessage<'_>,
 ) {
+    let span = trace_span!("Event Formation Message");
```

Comment:
This is a case that should just be done with a function annotation, not manually creating a span.